### PR TITLE
update to NSB v7

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@Particular/metrics-maintainers it seems the contracts for `IDurationProbe.Register` and `ISignalProbe.Register` have changed between v1 and v2. can u please point me to the upgrade guide for this change

File: packaging/nuget/NServiceBus.Metrics.PerformanceCounters.nuspec [hunk 1]
```diff
@@ -14,8 +14,8 @@
     <copyright>$copyright$</copyright>
     <tags>$tags$</tags>
     <dependencies>
-      <dependency id="NServiceBus" version="[6.2.0, 7.0.0)" />
-      <dependency id="NServiceBus.Metrics" version="[1.1.0, 2.0.0)" />
+      <dependency id="NServiceBus" version="[7.0.0-*, 8.0.0)" />
+      <dependency id="NServiceBus.Metrics" version="[2.0.0-*, 3.0.0)" />
```
Comment (andreasohlund) [inline packaging/nuget/NServiceBus.Metrics.PerformanceCounters.nuspec]: Is there a reason to stay with the old packaging?

**Comment by tmasternak** [comment]
> it seems the contracts for IDurationProbe.Register and ISignalProbe.Register have changed between v1 and v2.

@SimonCropp I'm probably missing something but those public API types from `NServiceBus.Metrics`.

File: packaging/nuget/NServiceBus.Metrics.PerformanceCounters.nuspec (hunk 1 above)
Comment (SimonCropp) [reply packaging/nuget/NServiceBus.Metrics.PerformanceCounters.nuspec]: no. but i wasnt sure how much to scope in this PR. so decided to leave it as nsb 7 for now

**Comment by SimonCropp** [comment]
@tmasternak yes those public APIs are from `NServiceBus.Metrics` which is owned by @Particular/metrics-maintainers so i would assume the upgrade guide would be under their responsibility

File: packaging/nuget/NServiceBus.Metrics.PerformanceCounters.nuspec (hunk 1 above)
Comment (andreasohlund) [reply packaging/nuget/NServiceBus.Metrics.PerformanceCounters.nuspec]: Makes sense, I have a call with @seanfarmar today to get him up to speed on the new packaging

**Comment by SimonCropp** [comment]
@Particular/metrics-maintainers build is green. can this be merged and pushed to myget?

**Comment by seanfarmar** [comment]
@tmasternak @Scooletz I'd be happy to do this with one of you guiding me

**Pull request merged**